Hmm, the test fails on MacOS because `make test_install` uses default compiler not the compiler with which is GridKit built:
```
$ make test_install
/.../local/gridkit-gcc
/.../local/gridkit-gcc/share/gridkit/examples/GridKitConsumer
-- The CXX compiler identification is AppleClang 17.0.0.17000319
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found BLAS: /opt/local/lib/libopenblas.dylib
-- Found LAPACK: /opt/local/lib/libopenblas.dylib;/opt/local/lib/libopenblas.dylib
-- Configuring done (0.4s)
-- Generating done (0.0s)
-- Build files have been written to: /.../local/gridkit-gcc/share/gridkit/examples/GridKitConsumer/build
[ 50%] Building CXX object CMakeFiles/consumer.dir/consumer.cpp.o
[100%] Linking CXX executable consumer
Undefined symbols for architecture arm64:
  "AnalysisManager::Sundials::Ida<double, unsigned long>::runSimulation(double, int, std::__1::optional<std::__1::function<void (double)>>)", referenced from:
      _main in consumer.cpp.o
      _main in consumer.cpp.o
      _main in consumer.cpp.o
  "GridKit::PhasorDynamics::parseSystemModelData(std::__1::__fs::filesystem::path const&)", referenced from:
      _main in consumer.cpp.o
ld: symbol(s) not found for architecture arm64
c++: error: linker command failed with exit code 1 (use -v to see invocation)
make[6]: *** [consumer] Error 1
make[5]: *** [CMakeFiles/consumer.dir/all] Error 2
make[4]: *** [all] Error 2
make[3]: *** [examples/Consumer/CMakeFiles/test_install] Error 2
make[2]: *** [examples/Consumer/CMakeFiles/test_install.dir/all] Error 2
make[1]: *** [examples/Consumer/CMakeFiles/test_install.dir/rule] Error 2
make: *** [test_install] Error 2
mac150403:build-gcc 55y$
```

If I manually select the right compiler, the test fails because it does not find the input file:

```
$ CC=gcc CXX=g++ make test_install
/Users/55y/local/gridkit-gcc
/Users/55y/local/gridkit-gcc/share/gridkit/examples/GridKitConsumer
-- The CXX compiler identification is GNU 13.3.0
-- Checking whether CXX compiler has -isysroot
-- Checking whether CXX compiler has -isysroot - yes
-- Checking whether CXX compiler supports OSX deployment target flag
-- Checking whether CXX compiler supports OSX deployment target flag - yes
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/local/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found BLAS: /opt/local/lib/libopenblas.dylib
-- Found LAPACK: /opt/local/lib/libopenblas.dylib;/opt/local/lib/libopenblas.dylib
-- Configuring done (1.2s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/55y/local/gridkit-gcc/share/gridkit/examples/GridKitConsumer/build
[ 50%] Building CXX object CMakeFiles/consumer.dir/consumer.cpp.o
[100%] Linking CXX executable consumer
[100%] Built target consumer
-- Install configuration: ""
Test project /Users/55y/local/gridkit-gcc/share/gridkit/examples/GridKitConsumer/build
    Start 1: GridKitConsumer
1/1 Test #1: GridKitConsumer ..................***Failed    0.28 sec

ERROR: No input file found or provided.

Usage:
       TwoBusBasicJson <json-input-file>

Please provide a JSON input file as a positional command-line 
argument.

By default this example will look for "TwoBusBasic.json" in the 
current working directory and use that if found.



0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.28 sec

The following tests FAILED:
	  1 - GridKitConsumer (Failed)
Errors while running CTest
make[3]: *** [examples/Consumer/CMakeFiles/test_install] Error 8
make[2]: *** [examples/Consumer/CMakeFiles/test_install.dir/all] Error 2
make[1]: *** [examples/Consumer/CMakeFiles/test_install.dir/rule] Error 2
make: *** [test_install] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -31,6 +31,7 @@
 - Added `Logger` class.
 - Added `CliOptions` for simple command-line parsing.
 - Added public function to encapsulate use of nlohmann/json and magic_enum from users.
+- Added complete basic consumer example project used for testing installation
 
 ## v0.1
 

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -54,3 +54,5 @@ endif()
 if(GRIDKIT_ENABLE_ENZYME)
   add_subdirectory(Enzyme)
 endif()
+
+add_subdirectory(Consumer)

--- a/examples/Consumer/CMakeLists.txt
+++ b/examples/Consumer/CMakeLists.txt
@@ -33,10 +33,11 @@ endif()
 
 # Install CMakeLists.txt for example project
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/GridKitConsumer/CMakeLists.txt.in
-    ${CMAKE_CURRENT_BINARY_DIR}/GridKitConsumer/CMakeLists.txt)
+    ${CMAKE_CURRENT_BINARY_DIR}/GridKitConsumer/CMakeLists.txt
+    @ONLY)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/GridKitConsumer/CMakeLists.txt
   DESTINATION ${_rel_consumer_path})
 
 # Add target to build and test the example project after install
-add_custom_target(test_install
-    COMMAND ${_abs_consumer_path}/test.sh ${CMAKE_INSTALL_PREFIX})
+add_custom_target(test_install COMMAND
+    ${_abs_consumer_path}/test.sh ${CMAKE_INSTALL_PREFIX} ${CMAKE_CXX_COMPILER})

--- a/examples/Consumer/CMakeLists.txt
+++ b/examples/Consumer/CMakeLists.txt
@@ -1,0 +1,42 @@
+# Path where the consumer test code will be installed
+set(_rel_consumer_path ${GRIDKIT_EXAMPLES_INSTALL_ROOT}/GridKitConsumer)
+set(_abs_consumer_path ${CMAKE_INSTALL_PREFIX}/${_rel_consumer_path})
+
+# Install script as example of how to build and test consumer project
+install(PROGRAMS test.sh DESTINATION ${_rel_consumer_path})
+
+# Install consumer code and set link libraries
+set(_examples_dir ${CMAKE_SOURCE_DIR}/examples)
+if(GRIDKIT_ENABLE_SUNDIALS)
+  install(
+    FILES ${_examples_dir}/PhasorDynamics/Tiny/TwoBus/Basic/TwoBusBasicJson.cpp
+    DESTINATION ${_rel_consumer_path}
+    RENAME consumer.cpp)
+  install(
+    FILES
+      ${_examples_dir}/PhasorDynamics/Tiny/TwoBus/Basic/TwoBusBasic.hpp
+      ${_examples_dir}/PhasorDynamics/Tiny/TwoBus/Basic/TwoBusBasic.json
+    DESTINATION ${_rel_consumer_path})
+  list(APPEND GRIDKIT_CONSUMER_TEST_LINK_LIBRARIES
+    GRIDKIT::phasor_dynamics_components
+    GRIDKIT::solvers_dyn)
+else()
+  install(
+    FILES ${_examples_dir}/PowerElectronics/DistributedGeneratorTest/DGTest.cpp
+    DESTINATION ${_rel_consumer_path}
+    RENAME consumer.cpp)
+  list(APPEND GRIDKIT_CONSUMER_TEST_LINK_LIBRARIES
+    GRIDKIT::power_elec_disgen
+    GRIDKIT::power_elec_microline
+    GRIDKIT::power_elec_microload)
+endif()
+
+# Install CMakeLists.txt for example project
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/GridKitConsumer/CMakeLists.txt.in
+    ${CMAKE_CURRENT_BINARY_DIR}/GridKitConsumer/CMakeLists.txt)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/GridKitConsumer/CMakeLists.txt
+  DESTINATION ${_rel_consumer_path})
+
+# Add target to build and test the example project after install
+add_custom_target(test_install
+    COMMAND ${_abs_consumer_path}/test.sh ${CMAKE_INSTALL_PREFIX})

--- a/examples/Consumer/GridKitConsumer/CMakeLists.txt.in
+++ b/examples/Consumer/GridKitConsumer/CMakeLists.txt.in
@@ -1,0 +1,28 @@
+# Example of how to consume GridKit as a package via cmake
+# Below is a standard example of a CMakeLists.txt file that ultizies GridKit
+# See the Readme on how to build and install GridKie
+#-------------------------------------------------------------------------------
+cmake_minimum_required(VERSION 3.20)
+
+# Project name
+project(GridKitConsumer LANGUAGES CXX)
+# GridKit uses C++20 features in its public headers
+set(CMAKE_CXX_STANDARD 20)
+
+# Call find_package to get GridKit targets
+# NOTE: The GridKit installation prefix needs to be provided in either
+# GridKit_DIR or CMAKE_PREFIX_PATH
+find_package(GridKit REQUIRED)
+
+# Create an executable target from our example code
+add_executable(consumer consumer.cpp)
+# Link against a GridKit library target
+target_link_libraries(consumer
+    @GRIDKIT_CONSUMER_TEST_LINK_LIBRARIES@
+)
+
+# Create a test that can be run via ctest
+include(CTest)
+add_test(NAME GridKitConsumer
+    COMMAND consumer
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})

--- a/examples/Consumer/GridKitConsumer/CMakeLists.txt.in
+++ b/examples/Consumer/GridKitConsumer/CMakeLists.txt.in
@@ -25,4 +25,4 @@ target_link_libraries(consumer
 include(CTest)
 add_test(NAME GridKitConsumer
     COMMAND consumer
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+    WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})

--- a/examples/Consumer/test.sh
+++ b/examples/Consumer/test.sh
@@ -1,0 +1,33 @@
+#!/bin/bash
+
+# This script gets executed in `make test_install`. 
+
+# GridKit install prefix used for CMake dependency finding
+export GridKit_DIR=${1}
+echo ${GridKit_DIR}
+
+# Locate source of the consumer test app
+export INSTALL_BUILD_CONSUME=${GridKit_DIR}/share/gridkit/examples/GridKitConsumer
+echo ${INSTALL_BUILD_CONSUME}
+
+# Create build directory
+mkdir -p ${INSTALL_BUILD_CONSUME}/build
+
+rm -rf ${INSTALL_BUILD_CONSUME}/build/* &&
+
+# Configure consumer project
+cmake -B ${INSTALL_BUILD_CONSUME}/build \
+    -S ${INSTALL_BUILD_CONSUME} \
+    -DGridKit_DIR=${GridKit_DIR} &&
+
+# Build and install
+cmake --build ${INSTALL_BUILD_CONSUME}/build &&
+
+cmake --install ${INSTALL_BUILD_CONSUME}/build &&
+
+# Check
+cd ${INSTALL_BUILD_CONSUME}/build &&
+
+ctest --output-on-failure
+
+exit $?

--- a/examples/Consumer/test.sh
+++ b/examples/Consumer/test.sh
@@ -4,11 +4,13 @@
 
 # GridKit install prefix used for CMake dependency finding
 export GridKit_DIR=${1}
-echo ${GridKit_DIR}
+echo "GridKit_DIR: ${GridKit_DIR}"
+
+export COMPILER=${2}
 
 # Locate source of the consumer test app
 export INSTALL_BUILD_CONSUME=${GridKit_DIR}/share/gridkit/examples/GridKitConsumer
-echo ${INSTALL_BUILD_CONSUME}
+echo "Consumer directory: ${INSTALL_BUILD_CONSUME}"
 
 # Create build directory
 mkdir -p ${INSTALL_BUILD_CONSUME}/build
@@ -18,6 +20,7 @@ rm -rf ${INSTALL_BUILD_CONSUME}/build/* &&
 # Configure consumer project
 cmake -B ${INSTALL_BUILD_CONSUME}/build \
     -S ${INSTALL_BUILD_CONSUME} \
+    -DCMAKE_CXX_COMPILER=${COMPILER} \
     -DGridKit_DIR=${GridKit_DIR} &&
 
 # Build and install

--- a/examples/PhasorDynamics/Tiny/TwoBus/Basic/TwoBusBasicJson.cpp
+++ b/examples/PhasorDynamics/Tiny/TwoBus/Basic/TwoBusBasicJson.cpp
@@ -17,7 +17,6 @@
 #include <Model/PhasorDynamics/SystemModel.hpp>
 #include <Model/PhasorDynamics/SystemModelData.hpp>
 #include <Solver/Dynamic/Ida.hpp>
-#include <Utilities/Testing.hpp>
 
 int main(int argc, const char* argv[])
 {

